features/cellular/framework/AT/AT_CellularContext.cpp

```diff
@@ -458,7 +458,28 @@ nsapi_error_t AT_CellularContext::do_activate_context()
 
 nsapi_error_t AT_CellularContext::activate_ip_context()
 {
-    return find_and_activate_context();
+    nsapi_error_t ret = find_and_activate_context();
+#if !NSAPI_PPP_AVAILABLE
+    if (ret == NSAPI_ERROR_OK) {
+        pdpContextList_t params_list;
+        if (get_pdpcontext_params(params_list) == NSAPI_ERROR_OK) {
+            pdpcontext_params_t *pdp = params_list.get_head();
+            while (pdp) {
+                SocketAddress addr;
+                if (addr.set_ip_address(pdp->dns_secondary_addr)) {
+                    tr_info("DNS secondary %s", pdp->dns_secondary_addr);
+                    add_dns_server(addr, get_interface_name(NULL));
```

Comment:
get_interface_name(NULL) does not work. Function uses the buffer given in parameter, so it must be given.  E.g.

```
char ifn[5];
add_dns_server(addr, get_interface_name(ifn));
```